compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -461,7 +462,13 @@ fn check_opaque_meets_bounds<'tcx>(
             if tcx.def_kind(tcx.parent(def_id.to_def_id())) == DefKind::OpaqueTy => {}
         // Can have different predicates to their defining use
         hir::OpaqueTyOrigin::TyAlias { .. } => {
-            let wf_tys = ocx.assumed_wf_types(param_env, span, def_id);
+            let wf_tys = match ocx.assumed_wf_types(param_env, span, def_id) {
+                Ok(wf_tys) => wf_tys,
+                Err(errors) => {
+                    infcx.err_ctxt().report_fulfillment_errors(&errors);
+                    return;
+                }
+            };
```

Comment:
This pattern seems like it deserves an `infcx` helper that does all that and then gives an `Result<T, ErrorReported>`.